storage/src/metadata/storage.rs

```diff
@@ -187,10 +182,15 @@ impl<E: Clock + Storage + Metrics, K: Array> Metadata<E, K> {
     }
 
     /// Get a value from [Metadata] (if it exists).
-    pub fn get(&self, key: &K) -> Option<&Vec<u8>> {
+    pub fn get(&self, key: &K) -> Option<&V> {
         self.map.get(key)
     }
 
+    /// Get a mutable reference to a value from [Metadata] (if it exists).
+    pub fn get_mut(&mut self, key: &K) -> Option<&mut V> {
```

Comment:
This will remove a TON of encode/decode overhead for things we are modifying in-place.